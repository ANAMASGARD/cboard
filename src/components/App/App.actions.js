import { FINISH_FIRST_VISIT, UPDATE_DISPLAY_SETTINGS } from './App.constants';

export function updateDisplaySettings(payload = {}) {
  return {
    type: UPDATE_DISPLAY_SETTINGS,
    payload
  };
}

export function finishFirstVisit() {
  return {
    type: FINISH_FIRST_VISIT
  };
}
