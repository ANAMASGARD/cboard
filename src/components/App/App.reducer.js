import {
  FINISH_FIRST_VISIT,
  UPDATE_CONNECTIVITY,
  UPDATE_DISPLAY_SETTINGS
} from './App.constants';
import { LOGIN_SUCCESS, LOGOUT } from '../Account/Login/Login.constants';
import { DISPLAY_SIZE_STANDARD } from '../Settings/Display/Display.constants';

const initialState = {
  isConnected: true,
  isFirstVisit: true,
  displaySettings: {
    boardElementsSize: DISPLAY_SIZE_STANDARD,
    boardFontSize: DISPLAY_SIZE_STANDARD,
    settingsSize: DISPLAY_SIZE_STANDARD,
    pictogramsSize: DISPLAY_SIZE_STANDARD
  },
  userData: {}
};

function appReducer(state = initialState, action) {
  switch (action.type) {
    case UPDATE_DISPLAY_SETTINGS:
      const displaySettings = { ...state.displaySettings, ...action.payload };
      return {
        ...state,
        displaySettings
      };
    case UPDATE_CONNECTIVITY:
      return {
        ...state,
        isConnected: action.payload
      };
    case FINISH_FIRST_VISIT:
      return {
        ...state,
        isFirstVisit: false
      };
    case LOGIN_SUCCESS:
      return {
        ...state,
        isFirstVisit: false,
        userData: action.payload || {}
      };
    case LOGOUT:
      return {
        ...state,
        userData: {}
      };
    default:
      return state;
  }
}

export default appReducer;
