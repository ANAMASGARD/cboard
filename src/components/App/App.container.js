import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl, intlShape } from 'react-intl';

import registerServiceWorker from '../../registerServiceWorker';
import { showNotification } from '../Notifications/Notifications.actions';
import { isFirstVisit, isLogged } from './App.selectors';
import messages from './App.messages';
import App from './App.component';
import { DISPLAY_SIZE_STANDARD } from '../Settings/Display/Display.constants';

import {
  updateLoggedUserLocation,
  updateUnloggedUserLocation
} from '../App/App.actions';
import { isAndroid } from '../../cordova-util';
import API from '../../api';
export class AppContainer extends Component {
  static propTypes = {
    /**
     * App language direction
     */
    dir: PropTypes.string.isRequired,
    /**
     * @ignore
     */
    intl: intlShape.isRequired,
    /**
     * If 'true', user first visit
     */
    isFirstVisit: PropTypes.bool,
    /**
     * If 'true', user is logged in
     */
    isLogged: PropTypes.bool,
    /**
     * App language
     */
    lang: PropTypes.string.isRequired,
    displaySettings: PropTypes.object.isRequired
  };

  componentDidMount() {
    const localizeUser = () => {
      const {
        isLogged,
        updateLoggedUserLocation,
        updateUnloggedUserLocation
      } = this.props;

      if (isLogged) return updateLoggedUserLocation();
      return updateUnloggedUserLocation();
    };

    registerServiceWorker(
      this.handleNewContentAvailable,
      this.handleContentCached
    );

    localizeUser();

    if (isAndroid()) this.configInAppPurchasePlugin();
  }

  handleNewContentAvailable = () => {
    const { intl, showNotification } = this.props;
    showNotification(
      intl.formatMessage(messages.newContentAvailable),
      'refresh'
    ); //send the kind of notification.
  };

  handleContentCached = () => {
    const { intl, showNotification } = this.props;
    showNotification(intl.formatMessage(messages.contentIsCached));
  };

  configInAppPurchasePlugin = () => {
    window.CdvPurchase.store.validator = async function(product, callback) {
      try {
        const res = await API.postTransaction(product);
        if (!res.ok) throw res;

        callback(true, { res }); // success!
        //callback(true, { transaction: "your custom details" }); // success!
        // your custom details will be merged into the product's transaction field
      } catch (e) {
        if (!e.ok) {
          callback(false, {
            code: e.data.code, // **Validation error code
            error: {
              message: e.error.message
            }
          });
          return;
        }

        callback(false, 'Impossible to proceed with validation');
      }
    };

    // window.store
    //   .when('subscription')
    //   .approved(p => {
    //     console.log('Porverificar', p);
    //     p.verify();
    //   })
    //   .verified(p => {
    //     console.log('Verificado, cambiar estado', p);
    //     p.finish();
    //   })
    //   .owned(p => console.log(`you now own ${p.alias}`));
    window.CdvPurchase.store
      .when()
      .approved(transaction => {
        console.log('Porverificar', transaction);
        //window.CdvPurchase.store.verify(transaction);
      })
      .verified(receipt => {
        console.log('Verificado, cambiar estado', receipt);
        window.CdvPurchase.store.finish(receipt);
      });
  };

  render() {
    const {
      dir,
      isFirstVisit,
      isLogged,
      lang,
      displaySettings,
      isDownloadingLang
    } = this.props;

    const uiSize = displaySettings.uiSize || DISPLAY_SIZE_STANDARD;
    const fontSize = displaySettings.fontSize || DISPLAY_SIZE_STANDARD;
    const classes = [
      'Cboard__DisplaySettings',
      `Cboard__UISize__${uiSize}`,
      `Cboard__FontSize__${fontSize}`
    ];

    const htmlElement = document.getElementsByTagName('html')[0];
    htmlElement.className = classes.join(' ');

    return (
      <App
        dir={dir}
        isFirstVisit={isFirstVisit}
        isLogged={isLogged}
        lang={lang}
        dark={displaySettings.darkThemeActive}
        isDownloadingLang={isDownloadingLang}
      />
    );
  }
}

const mapStateToProps = state => ({
  dir: state.language.dir,
  isFirstVisit: isFirstVisit(state),
  isLogged: isLogged(state),
  lang: state.language.lang,
  displaySettings: state.app.displaySettings,
  isDownloadingLang: state.language.downloadingLang.isdownloading
});

const mapDispatchToProps = {
  showNotification,
  updateLoggedUserLocation,
  updateUnloggedUserLocation
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(AppContainer));
