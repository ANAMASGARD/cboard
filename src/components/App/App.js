import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';

import { initApp } from './App.actions';
import BoardContainer from '../Board';
import Notifications from '../Notifications';
import './App.css';

export class App extends Component {
  static propTypes = {
    locale: PropTypes.string,
    dir: PropTypes.string,
    initApp: PropTypes.func
  };

  componentDidMount() {
    const { initApp } = this.props;
    initApp();
  }

  render() {
    const { locale, dir } = this.props;

    return (
      <div className="App" onClick={() => {}}>
        <Helmet>
          <html lang={locale} dir={dir} />
        </Helmet>
        <BoardContainer />
        <Notifications />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    locale: state.language.locale,
    dir: state.language.dir
  };
};

const mapDispatchToProps = dispatch => {
  return {
    initApp: () => dispatch(initApp())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
