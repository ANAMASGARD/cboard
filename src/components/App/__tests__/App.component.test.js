import React from 'react';
import { shallow } from 'enzyme';

import App from '../App.component';

jest.mock('../App.messages', () => ({
  save: {
    id: 'cboard.components.App.save',
    defaultMessage: 'Save'
  },
  newContentAvailable: {
    id: 'cboard.components.App.newContentAvailable',
    defaultMessage: 'New content is available; please refresh.'
  },
  contentIsCached: {
    id: 'cboard.components.App.contentIsCached',
    defaultMessage: 'Content is cached for offline use.'
  }
}));

describe('App component tests', () => {
  const props = {
    dir: '/',
    isFirstVisit: true,
    isLogged: true,
    lang: 'es-ES',
    logout: () => {},
    user: {}
  };

  it('renders without crashing', () => {
    shallow(<App {...props} />);
  });
  it('Matches Snapshot', () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
