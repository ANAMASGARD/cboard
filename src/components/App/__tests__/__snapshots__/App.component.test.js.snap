// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests Matches Snapshot 1`] = `ShallowWrapper {}`;
